Migration Portal Upcoming Release (#2190)
EnterpriseDB/docs: product_docs/docs/migration_portal/3.5.0/04_mp_migrating_database/01_mp_schema_extraction.mdx

---
title: "Performing a schema extraction"

legacyRedirectsGenerated:
  # This list is generated by a script. If you need add entries, use the `legacyRedirects` key.
  - "/edb-docs/d/edb-postgres-migration-portal/user-guides/user-guide/3.0.1/mp_schema_extraction.html"
---

<div id="mp_schema_extraction" class="registered_link"></div>

You can use the Oracle’s Data Pump utility to extract metadata from your Oracle database to an SQL dump file. Migration Portal requires this file to be uploaded to assess the extracted schemas.

!!! Note

>     Migration Portal might become unresponsive for very large SQL files, depending on your system’s and browser’s resource availability. Migration Portal supports SQL files upto 1 GB. If your SQL file size exceeds 1 GB, extract fewer schemas at a time to reduce the SQL file size.

Reviewer: ```suggestion
    Migration Portal might become unresponsive for very large SQL files, depending on your system’s and browser’s resource availability. Migration Portal supports SQL files up to 1 GB. If your SQL file size exceeds 1 GB, extract fewer schemas at a time to reduce the SQL file size.
```

Author: Fixed.

## Extracting schemas to an SQL dump file

Note that Migration Portal only requires the metadata in the SQL dump file to assess the extracted schemas.

#### Prerequisites

> * If you plan on exporting schemas that are not your own, ensure that you are assigned the *DATAPUMP_IMP_FULL_DATABASE* role. Otherwise, you can only export your own schema.

Reviewer: If this is a db role use code formatting, not italics.

Author: Done.

 
   ```
The attributes and values in the above command specify the following options:  
   + `CONTENT=metadata_only` extracts only the metadata of the schemas.  
   + `INCLUDE=` specifies which database object types to include in the extraction. Append a comma-separated list of the [Supported object types](#supported-object-types) to only extract database object types that are supported by Migration Portal. 

4. From the command line, run the export command to generate a **db.dump** file. For example, to extract metadata information for `<Schema_1>`, `<Schema_2>`, `<Schema_3>`, and so on, run the following command:
 
   ```
   $ expdp <Username>@<ConnectIdentifier> DIRECTORY=DMPDIR SCHEMAS=<Schema_1>,<Schema_2>,<Schema_3> DUMPFILE=db.dump parfile=parameters.par
   ```

5. To generate a SQL file from the dump file, run the import command. 

   For example, to generate **YourSchemas.SQL** file from the db.dump file, enter the following command:
 
   ```
    $ impdp <Username>@<ConnectIdentifier> DIRECTORY=<directory_name> TRANSFORM=OID:n SQLFILE=YourSchemas.sql DUMPFILE=db.dump
   ```

### To extract all schemas in a database

!!! Note

>     Do not perform the following procedure from a user account that belongs to the excluded schemas list, see [Unsupported schemas](#unsupported-schemas). The impdp command will fail if the user account running the command is in the excluded list of schemas.

Reviewer: ```suggestion
    Do not perform the following procedure from a user account that belongs to the excluded schemas list, see [Unsupported schemas](#unsupported-schemas). The impdp command fails if the user account running the command is in the excluded list of schemas.
```

Author: Done.

2. Grant read, write permissions to the user running the Data Pump utility.

 `SQL> grant read,write on directory DMPDIR to <Username>;`

3. Before running the expdp command, create a file with a `.par` extension (for example: parameters1.par) on your server. Add attributes and values to the file as shown below: 

   ```
   $ cat parameters1.par

   CONTENT=metadata_only

   FULL=Y

   INCLUDE=SYNONYM,SEQUENCE,TYPE,TABLE,CONSTRAINT,PROCEDURE,FUNCTION,PACKAGE,INDEX,MATERIALIZED_VIEW,VIEW,TRIGGER,DB_LINK	
 
   ```
The attributes and values in the above command specify the following options:  
   + `CONTENT=metadata_only` extracts only the metadata of the schemas.  
   + `INCLUDE=` specifies which database object types to include in the extraction. Append a comma-separated list of the [Supported object types](#supported-object-types) to only extract database object types that are supported by Migration Portal.
   + `FULL=Y` extracts all schemas in the database.

4. From the command line, run the export command to generate a **db.dump** file, as follows:
 
    ```
    $ expdp <Username>@<ConnectIdentifier> DIRECTORY=DMPDIR DUMPFILE=db.dump parfile=parameters1.par
    ```

> 5. Before running the impdp command, create a parameter file with a `.par` extension (for example: parameters2.par) on your server. Add attributes and values to the file as shown below:

Reviewer: ```suggestion
5. Before running the impdp command, create a parameter file with a `.par` extension (for example, parameters2.par) on your server. Add attributes and values to the file:
```

Author: @prashantagar rightly pointed out that the command shows the contents of the file and does not add the contents. 

@drothery-edb i know we usually do not like using the word "below", so I'll replace it "as follows"

   ('ANONYMOUS','APEX_PUBLIC_USER','APEX_030200','APEX_040000','APEX_040200','APPQOSSYS','AUDSYS','CTXSYS','DMSYS','DBSNMP',
   'DBSFWUSER','DEMO','DIP','DMSYS', 'DVF','DVSYS','EXFSYS','FLOWS_FILES','FLOWS_020100', 'FRANCK','GGSYS','GSMADMIN_INTERNAL',
   'GSMCATUSER','GSMROOTUSER','GSMUSER','LBACSYS','MDDATA','MDSYS','MGMT_VIEW','OJVMSYS','OLAPSYS','ORDPLUGINS','ORDSYS',
   'ORDDATA','OUTLN','ORACLE_OCM','OWBSYS','OWBYSS_AUDIT','PDBADMIN','RMAN','REMOTE_SCHEDULER_AGENT','SI_INFORMTN_SCHEMA',
   'SPATIAL_CSW_ADMIN_USR','SPATIAL_WFS_ADMIN_USR','SQLTXADMIN','SQLTXPLAIN','SYS$UMF','SYS','SYSBACKUP','SYSDG','SYSKM',
   'SYSRAC','SYSTEM','SYSMAN','TSMSYS','WKPROXY','WKSYS','WK_TEST','WMSYS','XDB','XS$NULL'))"
   ```
   The attributes and values in the above command specify the following options:
   + `EXCLUDE` specifies which schemas to exclude. Append a list of the [Unsupported schemas](#unsupported-schemas) to ensure that the Data Pump utility *does not* extract any system schemas (including schemas beginning with **pg_**).

6. To generate a SQL file from the dump file, run the import command. 

>    For example, to generate **YourSchemas.SQL** file from the db.dump file, enter the following command:

Reviewer: ```suggestion
   For example, to generate **YourSchemas.SQL** file from the db.dump file, enter:
```

Author: Done.

 
   ```
    $ impdp <Username>@<ConnectIdentifier> DIRECTORY=<directory_name> TRANSFORM=OID:n SQLFILE=YourSchemas.sql DUMPFILE=db.dump
   ```

> ## Supported schemas and objects

Reviewer: ```suggestion
## Schemas and objects support
```

Author: Fixed.

|                       |                     |                   |                    |                         |
| :-------------------- | :------------------ | :---------------- | :----------------- | :---------------------- |
| ANONYMOUS	| APEX_PUBLIC_USER	| APEX_030200	| APEX_040000	| APEX_040200	 |
| APPQOSSYS	| AUDSYS	| CTXSYS	| DMSYS	| DBSNMP	 |
| DBSFWUSER	| DEMO	| DIP	| DMSYS	| DVF	 |
| DVSYS	| EXFSYS	| FLOWS_FILES	| FLOWS_020100	| FRANCK	 |
| GGSYS	| GSMADMIN_INTERNAL	| GSMCATUSER	| GSMROOTUSER	| GSMUSER	 |
| LBACSYS	| MDDATA	| MDSYS	| MGMT_VIEW	| OJVMSYS	 |
| OLAPSYS	| ORDPLUGINS	| ORDSYS	| ORDDATA	| OUTLN	 |
| ORACLE_OCM	| OWBSYS	| OWBYSS_AUDIT	| PDBADMIN	| RMAN	 |
| REMOTE_SCHEDULER_AGENT	| SI_INFORMTN_SCHEMA	| SPATIAL_CSW_ADMIN_USR	| SPATIAL_WFS_ADMIN_USR	| SQLTXADMIN	 |
| SQLTXPLAIN	| SYS$UMF	| SYS	| SYSBACKUP	| SYSDG	 |
| SYSKM	| SYSRAC	| SYSTEM	| SYSMAN	| TSMSYS	 |
| WKPROXY	| WKSYS	| WK_TEST	| WMSYS	| XDB	 |
| XS$NULL	| 				 |	

!!! Note
    EDB Postgres Advanced Server does not support schema names starting with **pg_**, which should also be excluded while extracting from Oracle.

> ### Supported object types

Reviewer: I'd list the supported types first to align with the section title and intro sentence (and it is generally better to start with the positive case)
